### ISSUE TITLE
NUnit tests are executed in a 32-bit process

### ISSUE THREAD

**Issue by rolfbjarne** [mono/monodevelop#6961]
Repro:

* Create a new NUnit Library Project.
* Run the following test:

```csharp
[Test ()]
public void TestCase ()
{
	Assert.AreEqual (8, IntPtr.Size);
}
```

Result: test failure

> Expected: 8
> But was: 4

32-bit processes won't work on macOS 10.15 (which will come out this fall), so the earlier this is fixed the better.

<details>
<summary>Version 7.7.2 (build 21)</summary>
Visual Studio Enterprise 2017 for Mac
Version 7.7.2 (build 21)
Installation UUID: 3de793cc-1bea-4b7e-9e90-bb6747c5ca7e
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.1.178 (master / eeaeb7e6)

	Package version: 518000239

Mono Framework MDK
Runtime:
	Mono 5.18.0.239 (2018-08/0d988bc696e) (64-bit)
	Package version: 518000239

NuGet
Version: 4.8.0.5385

.NET Core
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.2.0
	2.0.0
SDK: /usr/local/share/dotnet/sdk/2.2.101/Sdks
SDK Versions:
	2.2.101
	2.0.0
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.18.0/lib/mono/msbuild/15.0/bin/Sdks

Xamarin.Profiler
'/Applications/Xamarin Profiler.app' not found

Xamarin.Android
Version: 8.3.0.19 (Visual Studio Enterprise)
Android SDK: Not found

Android Device Manager
Version: 7.7.1.0
Hash: 06ceaea1

Apple Developer Tools
Xcode 10.1 (14460.46)
Build 10B61

Xamarin.Mac
Version: 5.7.0.49 (Visual Studio Enterprise)
Hash: cdecef15f
Branch: issue-5447
Build date: 2019-01-22 09:08:31+0100

Xamarin.iOS
Version: 12.7.0.49 (Visual Studio Enterprise)
Hash: cdecef15f
Branch: issue-5447
Build date: 2019-01-22 09:08:31+0100

Xamarin Inspector
Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

Build Information
Release ID: 707020021
Git revision: f1fea53df9eb0bb5890a9563c0d7ea7b03922144
Build date: 2018-12-14 18:37:35+00
Build branch: release-7.7
Xamarin extensions: 9c5cea335e6a9ce4ccfde31c4aa06c25a3576085

Operating System
Mac OS X 10.13.6
Darwin 17.7.0 Darwin Kernel Version 17.7.0
    Fri Nov  2 20:43:16 PDT 2018
    root:xnu-4570.71.17~1/RELEASE_X86_64 x86_64
</details>

> VS bug [#890092](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/890092)

**Comment by mrward** [comment]
I think this has been addressed here - https://github.com/mono/monodevelop/pull/6911 //cc @mkrueger

**Comment by rolfbjarne** [comment]
This now works if I build the test assembly with /platform:x64, but it does not work if I build the test assembly with /platform:anycpu.

**Comment by Therzok** [comment]
@rolfbjarne not sure if we should do anything. AnyCPU invokes `mono`, x86 -> `mono32`, x64` -> `mono64`.

I suppose we should have mono change mono to default to 64bit?

**Comment by rolfbjarne** [comment]
mono already defaults to 64-bit:

```
mono --version
Mono JIT compiler version 6.0.0.176 (2019-02/d75c142a464 Fri Apr 12 17:31:22 EDT 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
```

**Comment by Therzok** [comment]
@rolfbjarne Yup, I checked 5.18 too, found the bug, will PR a fix.

**Comment by Therzok** [comment]
https://github.com/mono/monodevelop/pull/7516

**Comment by viniciusjarina** [comment]
Also, doesn't work if you start the debugger, even if the Assembly is `x64` @Therzok

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-16] Comment by **xamarinc**



Fixed in version 8.2.0.261 (master)



Pull Request #7516 merged by: Lluis Sanchez  
 Author: mono  
 Commit: ad32b280b6e7d2fdfa1ba99920d72da11c4d890d (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **xamarinc**



Fixed in version 8.1.0.2444 (release-8.1)



Pull Request #7517 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 3d8eff194eb0513555f9b6219f098659e56353de (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-20] Comment by **xamarinc**



Fixed in version 8.2.0.261 (master)



Pull Request #7516 merged by: Lluis Sanchez  
 Author: mono  
 Commit: ad32b280b6e7d2fdfa1ba99920d72da11c4d890d (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-21] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #7516 merged by: Lluis Sanchez  
 Author: mono  
 Commit: ad32b280b6e7d2fdfa1ba99920d72da11c4d890d (xamarin/monodevelop)
